block/provider/create_volume.go

```diff
@@ -121,21 +122,23 @@ func validateVolumeRequest(volumeRequest *provider.Volume, clusterVolumeLabel st
 		return resourceGroup, iops, userError.GetUserError("InvalidVolumeName", nil, *volumeRequest.Name)
 	}
 
-	// Capacity should not be empty
-	if volumeRequest.Capacity == nil {
-		return resourceGroup, iops, userError.GetUserError("VolumeCapacityInvalid", nil, nil)
-	} else if *volumeRequest.Capacity < minSize {
-		return resourceGroup, iops, userError.GetUserError("VolumeCapacityInvalid", nil, *volumeRequest.Capacity)
-	}
-
 	// Read user provided error, no harm to pass the 0 values to RIaaS in case of tiered profiles
 	if volumeRequest.Iops != nil {
 		iops = ToInt64(*volumeRequest.Iops)
 	}
 	if volumeRequest.VPCVolume.Profile == nil {
 		return resourceGroup, iops, userError.GetUserError("VolumeProfileEmpty", nil)
 	}
-	if volumeRequest.VPCVolume.Profile.Name != customProfile && iops > 0 {
+
+	// Capacity should not be empty
+	if volumeRequest.Capacity == nil {
+		return resourceGroup, iops, userError.GetUserError("VolumeCapacityInvalid", nil, nil)
+	} else if volumeRequest.VPCVolume.Profile.Name != sdpProfile && *volumeRequest.Capacity < minSize {
```

Comment:
just suggestion to revert the condition like *volumeRequest.Capacity < minSize && volumeRequest.VPCVolume.Profile.Name != sdpProfile, and put in separate condition don't combine with if condition